// react-form-elements/lib/textbox/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireWildcard(require("react"));

var _classnames = _interopRequireDefault(require("classnames"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _useFormElement = _interopRequireDefault(require("../utils/use-form-element"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = Object.defineProperty && Object.getOwnPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : {}; if (desc.get || desc.set) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } } newObj.default = obj; return newObj; } }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

/**
 * TextBox Component.
 *
 */
var TextBox = (0, _react.forwardRef)((_ref, ref) => {
  var {
    name,
    initialValue,
    label,
    type,
    className,
    labelClassName,
    inputClassName,
    children
  } = _ref,
      otherProps = _objectWithoutProperties(_ref, ["name", "initialValue", "label", "type", "className", "labelClassName", "inputClassName", "children"]);

  var {
    id,
    value,
    handleChange,
    inputRef
  } = (0, _useFormElement.default)(initialValue, ref);
  var labelStyleProp = labelClassName === '' ? {} : {
    className: labelClassName
  };
  var inputStyleProp = inputClassName === '' ? {} : {
    className: inputClassName
  };
  var hasLabel = label.length > 0;
  return _react.default.createElement("div", {
    className: (0, _classnames.default)("rfe-form__row", className)
  }, hasLabel && _react.default.createElement("label", _extends({
    htmlFor: id
  }, labelStyleProp), label || ''), _react.default.createElement("input", _extends({
    type: type,
    id: id,
    ref: inputRef,
    name: name,
    onChange: handleChange,
    value: value
  }, inputStyleProp, otherProps)), children);
});
TextBox.displayName = 'ReactFormElements(TextBox)';
var _default = TextBox;
exports.default = _default;
TextBox.propTypes = {
  name: _propTypes.default.string,
  initialValue: _propTypes.default.string,
  label: _propTypes.default.string,
  type: _propTypes.default.string,
  className: _propTypes.default.string,
  inputClassName: _propTypes.default.string,
  labelClassName: _propTypes.default.string
};
TextBox.defaultProps = {
  name: 'TextBox',
  initialValue: '',
  type: 'text',
  label: 'label',
  className: '',
  inputClassName: '',
  labelClassName: ''
};
