// react-form-elements/lib/utils/use-form-element.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = require("react");

var _useId = _interopRequireDefault(require("./use-id"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/**
 * useFormElement Hook.
 *
 */
var useFormElement = function useFormElement(initialValue) {
  var ref = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : (0, _react.createRef)();
  var id = (0, _useId.default)();
  var [value, setValue] = (0, _react.useState)(initialValue);

  var handleChange = e => {
    if (typeof initialValue === 'boolean') {
      return setValue(!value);
    }

    setValue(e.target.value);
  };

  var inputRef = (0, _react.useRef)();
  (0, _react.useImperativeHandle)(ref, () => ({
    clear: () => setValue(''),
    reset: () => setValue(''),
    getValue: () => inputRef.current.value,
    getValidity: () => inputRef.current.validity,
    isValid: () => inputRef.current.validity.valid
  }));
  return {
    id,
    value,
    inputRef,
    handleChange
  };
};

var _default = useFormElement;
exports.default = _default;
